front/lib/models/assistant/conversation.ts

```diff
@@ -331,6 +333,79 @@ AgentMessage.init(
   }
 );
 
+export class AgentMessageFeedback extends Model<
+  InferAttributes<AgentMessageFeedback>,
+  InferCreationAttributes<AgentMessageFeedback>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare agentConfigurationId: ForeignKey<AgentConfiguration["id"]>;
+  declare agentConfigurationVersion: number;
+  declare agentConfigurationVersionDate: string;
+  declare agentMessageId: ForeignKey<AgentMessage["id"]>;
+  declare userId: ForeignKey<User["id"]>;
+
+  declare thumbDirection: AgentMessageFeedbackDirection;
+  declare content: string | null;
+}
+
+AgentMessageFeedback.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    agentConfigurationVersion: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      defaultValue: 0,
+    },
+    agentConfigurationVersionDate: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      defaultValue: 0,
+    },
+    thumbDirection: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+    content: {
+      type: DataTypes.TEXT,
+      allowNull: true,
+    },
+  },
+  {
+    modelName: "agent_message_feedback",
+    sequelize: frontSequelize,
+  }
+);
+
+AgentMessageFeedback.belongsTo(AgentConfiguration, {
+  as: "agentConfiguration",
+  foreignKey: { name: "agentConfigurationId", allowNull: false },
+});
+AgentMessageFeedback.belongsTo(AgentMessage, {
+  as: "agentMessage",
+  foreignKey: { name: "agentMessageId", allowNull: false },
+});
+AgentMessageFeedback.belongsTo(User, {
+  as: "user",
+  foreignKey: { name: "userId", allowNull: false },
+});
+
```

Comment:
I think we also need the opposite relation and enforce the onDelete: 'RESTRICT' (at least on the relation which have a resource like user_resource , and implement the cascade in `UserResource.delete` )

```
User.hasMany(AgentMessageFeedback, {
  onDelete: "RESTRICT",
});
```

Reply:
➕ We don't use the default value "ON DELETE CASCADE" anymore.